### PR TITLE
Enhancement: Supports adding third-party dependencies from git repo with the version field

### PR DESCRIPTION
**Pull request by Manoramsharma**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix : #266 

#### 2. What is the scope of this PR (e.g. component or file name):
pkg/client/client.go

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [x] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

I simply changed the logic of AddDepToPkg function to handle version field at the time of adding the dependencies to the pkg. 
#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

I directly modified the test_data for testing the unit tests. No need to make any other changes in unit test.

image

### CODE REVIEW

**Comment by Manoramsharma** [comment]
Hi @Peefy I was getting conflicts in the PR so here I have opened another one. But I still see the unit tests failing which are passing in my local see :
Screenshot 2024-08-02 at 7 09 39 PM

cc: @zong-zhe

File: pkg/client/test_data/add_with_git_tag/test_pkg_win/kcl.mod.lock.expect [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[dependencies]
+  [dependencies.flask-demo-kcl-manifests]
+    name = "flask-demo-kcl-manifests"
+    full_name = "flask_manifests_0.0.1"
```
[the remark this answers is not in the document]
Comment (Manoramsharma) [reply pkg/client/test_data/add_with_git_tag/test_pkg_win/kcl.mod.lock.expect:4]:
```
// GenDepFullName will generate the full name of a dependency by its name and version
// based on the '<package_name>_<package_tag>' format.
func (dep *Dependency) GenDepFullName() string {
    dep.FullName = fmt.Sprintf(PKG_NAME_PATTERN, dep.Name, dep.Version)
    return dep.FullName
}
```

Can this issue be resolved by changing the function mentioned above?

[the remark this answers is not in the document]
Comment (zong-zhe) [reply pkg/client/test_data/add_with_git_tag/test_pkg_win/kcl.mod.lock.expect:4]: Changes to this section may affect other features at this stage, resulting in more test failures and more work. And this part of the work will be meaningless after https://github.com/kcl-lang/kpm/issues/384 done. Therefore, we can try to stop this PR temporarily. Finish this PR after you finish https://github.com/kcl-lang/kpm/issues/384

[the remark this answers is not in the document]
Comment (Manoramsharma) [reply pkg/client/test_data/add_with_git_tag/test_pkg_win/kcl.mod.lock.expect:4]: I got your point now @zong-zhe

**Comment by Manoramsharma** [comment]
> The CI failed. cc @Manoramsharma



> The CI failed. cc @Manoramsharma

@zong-zhe told me to stop this PR until the #384 will be completed.

**Pull request closed without merging**